### PR TITLE
[Apicurio] Please merge generated API project

### PR DESCRIPTION
**Pull request by craicoverflow**
The Apicurio tool was used to generate an API project from an OpenAPI definition.  Please review this pull request, modify as necessary, and merge!

### CODE REVIEW

**Review by craicoverflow — COMMENTED** [review]
@Apicurio/developers there are a couple of changes here that I can't explain. This is a direct generation from Studio. I've pinged @carlesarnal and @jsenko in the two areas I want to highlight as they have made changes to these sections very recently.

File: common/src/main/java/io/apicurio/registry/rest/v2/beans/Limits.java [hunk 1]
```diff
@@ -27,8 +26,8 @@
     "maxPropertyValueSizeBytes",
     "maxArtifactLabelsCount",
     "maxLabelSizeBytes",
-    "maxArtifactNameLengthChars",
-    "maxArtifactDescriptionLengthChars",
+    "maxNameLengthChars",
+    "maxDescriptionLengthChars",
```
[the remark this answers is not in the document]
Comment (jsenko) [reply common/src/main/java/io/apicurio/registry/rest/v2/beans/Limits.java:30]: The schema in Studio that you used to generate code is using the older names of those two properties. I see that the OpenAPI spec in the code is using the correct names, so I think the schema in studio is missing this edit. I'm not sure why, but the solution would be to rename those two properties and generate the code again.

File: common/src/main/java/io/apicurio/registry/rest/v2/IdsResource.java [hunk 2]
```diff
@@ -97,5 +97,5 @@ Response getContentByGlobalId(@PathParam("globalId") int globalId,
   @Path("/globalIds/{globalId}/references")
   @GET
   @Produces("application/json")
-  List<ArtifactReference> referencesByGlobalId(@PathParam("globalId") Long globalId);
+  List<ArtifactReference> referencesByGlobalId(@PathParam("globalId") Object globalId);
```
[the remark this answers is not in the document]
Comment (carlesarnal) [reply common/src/main/java/io/apicurio/registry/rest/v2/IdsResource.java:100]: Very likely this is due to the type not being properly set in the OpenAPI definition. There was a problem with some references related endpoints not being present in the OpenAPI, I added them back manually a few days ago and I probably forgot to add the type. How do you want to proceed? We can add the type to the definition and regenerate or we can just add the type and revert the related changes in this PR.

File: common/src/main/java/io/apicurio/registry/rest/v2/beans/Limits.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (craicoverflow) [reply common/src/main/java/io/apicurio/registry/rest/v2/beans/Limits.java:30]: Great, that's fine and I can do that. I just wasn't sure if Studio was correct or not and did not want to overwrite incorrectly.

File: common/src/main/java/io/apicurio/registry/rest/v2/IdsResource.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (craicoverflow) [reply common/src/main/java/io/apicurio/registry/rest/v2/IdsResource.java:100]: I can add the type and re-generate. Thanks @carlesarnal

**Pull request closed without merging**